### PR TITLE
Integration test review

### PR DESCRIPTION
**Pull request by marota**
Changes in load, renewable and dispatch generation were first commented as minimally as possible to get back to a version with working integration tests. Then changes were uncommented and introduced again incrementally while checking proper behavior and evolution of integration tests.

### CODE REVIEW

File: chronix2grid/generation/consumption/consumption_utils.py [hunk 1]
```diff
@@ -24,7 +24,7 @@ def compute_loads(loads_charac, temperature_noise, params, load_weekly_pattern,
     first_dow_chronics = datetime.strptime(load_weekly_pattern["datetime"].iloc[1], "%Y-%m-%d %H:%M:%S").weekday()
     # + (calendar.isleap(start_day.year) if start_day.month >= 3 else 0)
     day_lag = (first_dow_chronics - start_day_of_week) % 7
-    day_lag = 6  # this is only TRUE if you simulate 2050 !!!
+    day_lag = 0#6  # this is only TRUE if you simulate 2050 !!!
```
Comment (BDonnot) [inline chronix2grid/generation/consumption/consumption_utils.py:27]: Would it be possible to have this as argument of this function, with a possible default value of 0 if you'd like.
Even better would be to have it in a config file somewhere.

We really need it for wcci 2022

File: requirements.txt [hunk 2]
```diff
@@ -9,19 +9,19 @@ json5>=0.9.3
 jupyter>=1.0.0
 matplotlib>=3.1.3
 numba>=0.48.0
-numpy>=1.18.3
-pandas>=1.0.3
+numpy==1.18.3
+pandas==1.0.5
```
Comment (BDonnot) [inline requirements.txt:13]: These are super super old version. It's probably required to introduce compatibility with most recent versions !

**Review by BDonnot — COMMENTED** [review]
I think the tests should be done with numpy >= 1.20 (at least)

Also, it's totally fine to set the package versions in the "requirements.txt" but not in the config.py and the "==" should be replaced by ">="

File: chronix2grid/generation/consumption/consumption_utils.py (hunk 1 above)
Comment (marota) [reply chronix2grid/generation/consumption/consumption_utils.py:27]: I agree we should address this issue https://github.com/BDonnot/ChroniX2Grid/issues/45 and improvement. 

This should be developed through a next branch and PR, as this one was only to make sure that all tests were well-aligned without evolution in functionnalities + continuously run through CircleCI

File: requirements.txt (hunk 2 above)
Comment (marota) [reply requirements.txt:13]: I agree, should also be addressed through a next branch and PR as "upgrading required package versions". This one was only to make sure that all tests were well-aligned without evolution in functionnalities + continuously run through CircleCI

File: setup.py [hunk 3]
```diff
@@ -34,17 +33,17 @@
                         "json5>=0.9.3",
                         "jupyter>=1.0.0",
                         "matplotlib>=3.1.3",
-                        "numpy>=1.18.3",
-                        "pandas>=1.0.5",
+                        "numpy==1.18.3",
+                        "pandas==1.0.5",
```
[the remark this answers is not in the document]
Comment (marota) [reply setup.py]: Right, so this was done in the last commit with test passing. 
For "numpy >= 1.20" this should be done through the suggested follow-up PR above

File: setup.py [hunk 4]
```diff
@@ -24,7 +24,6 @@
       include_package_data=True,
       install_requires=[
                         "click>=7.1.1",
-                        "pypsa",
```
Comment (BDonnot) [inline setup.py:27]: Is there a reason why pypsa is not on the setup.py ? It's mandatory to use the package, no ? (beside, pypsa license is now ok)

**Review by BDonnot — COMMENTED** [review]
Maybe reintroducing pypsa in the setup.py should be considered, beside that it looks good :-)

**Comment by BDonnot** [comment]
I'll merge it as it appears to be ok

**Pull request merged**